camelot/admin/filter_admin.py

```diff
@@ -0,0 +1,61 @@
+"""
+Admin classes for Filter strategies
+"""
+import copy
+import logging
+
+from camelot.core.utils import ugettext_lazy as _
+
+from .action import list_filter
+from .action.list_action import FilterValue
+from .object_admin import ObjectAdmin
+from ..view.controls import delegates
+
+LOGGER = logging.getLogger('camelot.admin.filter_admin')
+
+class FilterValueAdmin(ObjectAdmin):
+
+    verbose_name = _('Filter')
+    list_display = ['value_1', 'value_2']
+    field_attributes = {
+        'value_1': {'editable': True, 'choices': lambda o: o.strategy.choices},
+        'value_2': {'editable': True, 'visible': False},
+    }
+
+    def __init__(self, app_admin, entity):
+        assert issubclass(entity, FilterValue), '{} is not a FilterValue class'.format(entity)
+        super().__init__(app_admin, entity)
+
+    def get_name(self):
+        return self.entity.filter_strategy.name
+
+# Create and register filter value classes and related admins for each filter strategy
+# that has not got one registered already.
+for strategy_cls, delegate in [
+    (list_filter.StringFilter,  delegates.PlainTextDelegate),
+    (list_filter.BoolFilter,    delegates.BoolDelegate),
+    (list_filter.DateFilter,    delegates.DateDelegate),
+    (list_filter.DecimalFilter, delegates.FloatDelegate),
+    (list_filter.IntFilter,     delegates.IntegerDelegate),
+    (list_filter.TimeFilter,    delegates.TimeDelegate),
+    (list_filter.RelatedFilter, delegates.PlainTextDelegate),
+    ]:
+    try:
+        FilterValue.get_filter_value(strategy_cls)
```

Comment:
kunnen we die value classes nt gewoon in loop aanmaken, is er een reden voor die try/except ?

Reply:
Mja, op dit moment laat ik toe om een gecustomizeerde FilterValue class te definiëren als inner class van een strategy. Die try except is om er voor te zorgen dat er enkel een wordt aangemaakt als er nog geen bestaat.